tests/tape/tapes/test_tape.py

```diff
@@ -840,6 +840,26 @@ def test_expand_tape_multiple_wires_non_commuting(self, ret):
         with pytest.raises(qml.QuantumFunctionError, match="Only observables that are qubit-wise"):
             tape.expand(expand_measurements=True)
 
+    def test_is_sampled_reserved_after_expansion(self, monkeypatch, mocker):
```

Comment:
(thinking aloud) I think this test looks good, but I'm contemplating whether this is too broad of a test (not unit-test-like enough), but I'm not sure how to do it otherwise.

Reply:
Yeah, think that's a fair point. One thing I found challenging with creating more of a unit test is that a tape is created and expanded in `QNode.construct` along with other updates (perhaps later down the line we could think about separating these into different methods).

Reply:
Without being to familiar with writing tests, etc. I would say a good option could simply be to be better at separating the unit tests and the integration tests (and this could perhaps be part of the latter, right?). Then no unit test would be needed for this case at all.

Reply:
Hmm, it seems that integration tests were not too separated so far for tapes. Might just leave it here for now and we could move it if we start the separation into a new file.